src/RamIndexAnalysis.h

```diff
@@ -34,16 +34,160 @@
 #include <utility>
 #include <vector>
 
-#define RIA_NIL 0
-#define RIA_INF -1
-
 // define if enable unit tests
 #define M_UNIT_TEST
 
 namespace souffle {
 
 class RamTranslationUnit;
 
+/** search signature of a RAM operation; each bit represents an attribute of a relation.
+ * A one represents that the attribute has an assigned value; a zero represents that
+ * no value exists (i.e. attribute is unbounded) in the search. */
+class SearchSignature {
+public:
+    explicit SearchSignature(size_t arity, uint64_t mask) : bits(arity, false) {
```

Comment:
Add `default`ed copy/move ctors/assigners?

Reply:
Remove `arity` argument, the size of the storage should be an internal detail.

Reply:
I feel like the arity is intrinsic to the nature of the SearchSignature representing arity constraints (a constraint for each attribute). Given that we now have a vector<AttributeConstraint> perhaps the rarity in the constructor makes sense now?